snuba/cli/__init__.py

```diff
@@ -17,8 +29,13 @@ def main() -> None:
     `OoO'  o   O `OoO'o `OoO' `OoO'o"""
 
 
-for loader, module_name, is_pkg in walk_packages(__path__, __name__ + "."):
-    module = __import__(module_name, globals(), locals(), ["__name__"])
-    cmd = getattr(module, module_name.rsplit(".", 1)[-1])
-    if isinstance(cmd, click.Command):
-        main.add_command(cmd)
+with sentry_sdk.start_transaction(op="snuba_init", name="Snuba CLI Initialization"):
+    for loader, module_name, is_pkg in walk_packages(__path__, __name__ + "."):
```

Comment:
This is likely to initialize a transaction on consumers as well.
We cannot afford to do that without doing a throughout test before and without testing locally what gets produced there. We know snuba produces billions of locally dropped transactions which likely come from consumers.
So please check that consumer does not produce transactions

Reply:
Tested behavior of consumers locally - all consumer startups create a transaction which is finished and sent to Sentry before message processing begins. There should only be 1 transaction per consumer start (or restart). The added spans may make it into any new transactions created around Dataset component factory initialization but even this initialization should only happen once per consumer startup.